src/main/java/org/variantsync/diffdetective/variation/NodeType.java

```diff
@@ -52,7 +57,36 @@ public static NodeType fromName(final String name) {
     }
 
     /**
-     * Returns the number of bits required for storing {@link ordinal}.
+     * Creates a NodeType from an AnnotationType.
+     * <p>
+     * All AnnotationType variants except for 'Endif' are supported.
+     * There is no valid representation for 'Endif' annotations. Thus, the method throws an IllegalArgumentException
+     * if it is given an 'Endif'.
+     * </p>
+     *
+     * @param annotationType a variant of AnnotationType
+     * @return The NodeType that fits the given AnnotationType
+     */
+    public static NodeType fromAnnotationType(final AnnotationType annotationType) {
+        switch (annotationType) {
+            case If -> {
+                return NodeType.IF;
+            }
+            case Elif -> {
+                return NodeType.ELIF;
+            }
+            case Else -> {
+                return NodeType.ELSE;
+            }
+            case None -> {
+                return NodeType.ARTIFACT;
+            }
+            default -> throw new IllegalArgumentException(annotationType + "has no NodeType counterpart");
+        }
```

Comment:
If you want to be fancy, you might want to use a [switch expression](https://docs.oracle.com/en/java/javase/17/language/switch-expressions.html#GUID-BA4F63E3-4823-43C6-A5F3-BAA4A2EF3ADC) but this looks fine as is. :)

Reply:
What do you mean? It already is a switch expression?

Reply:
It is a switch statement. :) Thought, you might be interested in some new fancy Java syntax. You could write

```java
return switch(annotationType) {
    case If -> NodeType.If;
    case Else -> NodeType.Else;
    ...
};
```